CalibTracker/SiStripChannelGain/src/APVGainHelpers.cc

```diff
@@ -106,89 +106,91 @@ int APVGain::subdetectorPlane(const std::string& tag) {
+      (*it).printAll();
     }
-    return found;
+    it++;
+  }
+  return found;
 }
 
 
 
 std::vector<std::pair<std::string,std::string>> 
 APVGain::monHnames(std::vector<std::string> VH, bool allPlanes, const char* tag) {
 
-    std::vector<std::pair<std::string,std::string>> out;
-    int re = (allPlanes)? 34 + VH.size() : VH.size();
-    out.reserve( re );
-
-
-    std::string Tag = tag;
-    if (Tag.length())  Tag = "__" + Tag;
-
-    std::string h_tag = "";
-    std::string h_tit = "";
-
-    if (allPlanes) {
-        // Names of monitoring histogram for TIB layers
-        int TIBlayers = 4;  //number of TIB layers.
-        for(int i=1; i<=TIBlayers; i++) {
-            h_tag = "TIB_layer_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-        // Names of monitoring histogram for TOB layers
-        int TOBlayers = 6;  //number of TOB layers
-        for(int i=1; i<=TOBlayers; i++) {
-            h_tag = "TOB_layer_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-        // Names of monitoring histogram for TID wheels
-        int TIDwheels = 3;  //number of TID wheels
-        for(int i=-TIDwheels; i<=TIDwheels; i++) {
-            if (i==0) continue;
-            if (i<0)  h_tag = "TIDminus_wheel_" + std::to_string(i) + Tag;
-            else      h_tag = "TIDplus_wheel_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-        // Names of monitoring histogram for TEC wheels
-        int TECwheels = 9;  //number of TEC wheels
-        for(int i=-TECwheels; i<=TECwheels; i++) {
-            if (i==0) continue;
-            if (i<0) h_tag = "TECminus_wheel_" + std::to_string(i) + Tag;
-            else     h_tag = "TECplus_wheel_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-    }
+  std::vector<std::pair<std::string,std::string>> out;
+  int re = (allPlanes)? 34 + VH.size() : VH.size();
+  out.reserve( re );
 
-    for(unsigned int i=0; i<VH.size();i++) {
-        h_tag = VH[i] + Tag;
-        h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-        out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-    }
+  std::string Tag = tag;
+  if (Tag.length())  Tag = "__" + Tag;
 
-    return out;
+  std::string h_tag = "";
+  std::string h_tit = "";
+
+  if (allPlanes) {
+    // Names of monitoring histogram for TIB layers
+    int TIBlayers = 4;  //number of TIB layers.
```

Comment:
could be `constexpr` (or can we retrieve this from the database?)
applies also for the other subdetectors below

Reply:
I made them `constexpr`. 
Of course, I agree it's ugly and this might be retrieved from the topology, but to make it general there is much more work needed to percolate it down from the event setup and adjust the other methods logic than just hardcode the n. of layers here. Anyway we don't foresee to use this for any phase-II production (yet). Unless you insist I would defer this to a follow-up PR (it exceeds the scope of this one)